### PR TITLE
Enable ConvTranspose op

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 1]
```diff
@@ -571,6 +571,49 @@ def Conv(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> Ngra
     return make_convolution_op(onnx_node, ng_inputs)
 
 
+def ConvTranspose(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> NgraphNode
+    """Calculate convolution transpose."""
+    if len(ng_inputs) == 3:
+        data, weights, bias = ng_inputs
+    elif len(ng_inputs) == 2:
```
Comment (arogowie-intel) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: What if number of inputs will be different? Consider adding `else` statement and give some warning/error.

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 2]
```diff
@@ -571,6 +571,49 @@ def Conv(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> Ngra
...
+        data, weights = ng_inputs
+        bias = ng.constant(0, dtype=get_dtype(data.get_element_type()))
+
+    strides = get_strides(onnx_node)
+    dilation = get_dilations(onnx_node)
+    padding_below, padding_above = get_pads(onnx_node)
+
+    output_padding = onnx_node.get_attribute_value('output_padding')
```
Comment (arogowie-intel) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: Ensure some default value or raise error if `output_padding` not present.

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 3]
```diff
@@ -571,6 +571,49 @@ def Conv(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> Ngra
...
+
+    weights_shape = list(weights.shape)
+    data_shape = list(data.shape)
+
+    num_spatial_dims = len(data.shape) - 2
+
+    data_dilation_strides = [1, 1]
+
```
Comment (arogowie-intel) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: Please, think about making your code spatially more compact, eg: removing superfluous empty lines.

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 4]
```diff
@@ -571,6 +571,49 @@ def Conv(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> Ngra
...
+    data_batch_shape = [1] * (num_spatial_dims + 2)
+    data_batch_shape[0] = data_shape[0]
+    data_batch_shape[1] = weights_shape[1]
+
+    for i in range(num_spatial_dims):
+        data_batch_shape[i + 2] = ((padding_below[i] + ((data_shape[i + 2] - 1) * strides[i] + 1) + output_padding[i]) -
```
Comment (arogowie-intel) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: Here is some magic. Please try adding some comment explaining the mathematical formula, or maybe some urls what is mapped into what.

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py (hunk 1 above)
Comment (tsocha) [reply ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: Usually we are not verify number of inputs and standard says there can be only 2 or 3 inputs.

File: tests/test_backend.py [hunk 5]
```diff
@@ -51,11 +51,6 @@
 
 # Ops
 # Missing ops
-# Missing op 'ConvTranspose' -> NGRAPH-1840
-backend_test.exclude('test_ConvTranspose2d_cpu')
-backend_test.exclude('test_ConvTranspose2d_no_bias_cpu')
-backend_test.exclude('test_operator_convtranspose_cpu')
-
```
Comment (arogowie-intel) [inline tests/test_backend.py]: Add some our UT, you may use examples from [doc](https://github.com/onnx/onnx/blob/master/docs/Operators.md#examples-16)

**Comment by postrational** [comment]
TESTNOW

File: tests/test_backend.py (hunk 5 above)
Comment (postrational) [reply tests/test_backend.py]: I don't think we need to add custom tests if there are ONNX tests covering `ConvTranspose`.

**Pull request merged**